### PR TITLE
association corrected

### PR DESCRIPTION
**Pull request by rubyboy-hub**
changes in design ,login part, registration part done.

### CODE REVIEW

File: app/views/users/sessions/new.html.erb [hunk 1]
```diff
@@ -1,29 +1,61 @@
...
-    <%= f.submit "Log in" %>
+    <%= f.submit "Sign in",class:"w-100 btn btn-lg btn-primary" %>
   </div>
 <% end %>
+</main>
 
 <%= render "users/shared/links" %>
+
+
+<style type="text/css">
```
Comment (neethu-satheesh) [inline app/views/users/sessions/new.html.erb:32]: @rubyboy-hub  Hi, The css code needs to be moved to corresponding stylesheet folder inside app/assets.  Normally, we don't write the css in erb files.

**Review by neethu-satheesh — CHANGES_REQUESTED** [review]
@rubyboy-hub   Hi Shanjoy, the association looks good. Added a review abt the css.

File: app/views/users/sessions/new.html.erb (hunk 1 above)
Comment (rubyboy-hub) [reply app/views/users/sessions/new.html.erb:32]: Okay , I will add the association and  change the respective css file  into a separate css file

File: app/models/user.rb [hunk 2]
```diff
@@ -1,5 +1,7 @@
 class User < ApplicationRecord
   belongs_to :role
+  belongs_to :standard
```
Comment (neethu-satheesh) [inline app/models/user.rb:3]: @rubyboy-hub  Hi, An intermediate table will have has_many :intermediate_table_infos

File: db/migrate/20210524121724_model.rb [hunk 3]
```diff
@@ -0,0 +1,4 @@
+class Model < ActiveRecord::Migration[6.1]
```
Comment (neethu-satheesh) [inline db/migrate/20210524121724_model.rb:1]: @rubyboy-hub  If this is not being used, lets remove it please

File: db/migrate/20210525093549_add_parent_to_user.rb [hunk 4]
```diff
@@ -0,0 +1,5 @@
+class AddParentToUser < ActiveRecord::Migration[6.1]
+  def change
+    add_column :users, :parent_id, :bigint,index:true
```
Comment (neethu-satheesh) [inline db/migrate/20210525093549_add_parent_to_user.rb:3]: @rubyboy-hub  If you want, you can have the foreign constraint  by adding another line for add_foreign_key

File: app/models/comment.rb [hunk 5]
```diff
@@ -0,0 +1,3 @@
+class Comment < ApplicationRecord
+    belongs_to: user
```
Comment (neethu-satheesh) [inline app/models/comment.rb:2]: @rubyboy-hub  Hi, Comment has from and to -> Do you have the latest code that reflects those changes as we discussed?

File: app/models/standard.rb [hunk 6]
```diff
@@ -0,0 +1,4 @@
+class Standard < ApplicationRecord
+    has_many :subjects 
+    has_many :users
```
Comment (neethu-satheesh) [inline app/models/standard.rb:3]: @rubyboy-hub  This is incorrect as per the ER diagram

File: app/models/standard.rb [hunk 7]
```diff
@@ -0,0 +1,4 @@
+class Standard < ApplicationRecord
+    has_many :subjects
```
Comment (neethu-satheesh) [inline app/models/standard.rb:2]: @rubyboy-hub  The ER dia does not show this association. Do you still want to use it?

Comment (rubyboy-hub) [reply app/models/standard.rb:2]: yes, i had insert an standard foreign key to subjects, so all standard wise subject can be inserted to subject table. When a user class related to a standard. All his/her subject can be seen using this.

File: app/models/standard.rb (hunk 6 above)
Comment (rubyboy-hub) [reply app/models/standard.rb:3]: Applied new er diagram

File: app/models/comment.rb (hunk 5 above)
Comment (rubyboy-hub) [reply app/models/comment.rb:2]: Its available in test28052021 
  belongs_to :from_user, class_name: 'User', foreign_key: :from
    belongs_to :to_user, class_name: 'User', foreign_key: :to

File: db/migrate/20210525093549_add_parent_to_user.rb (hunk 4 above)
Comment (rubyboy-hub) [reply db/migrate/20210525093549_add_parent_to_user.rb:3]: i had added separately

File: db/migrate/20210524121724_model.rb (hunk 3 above)
Comment (rubyboy-hub) [reply db/migrate/20210524121724_model.rb:1]: removed

File: app/models/user.rb (hunk 2 above)
Comment (rubyboy-hub) [reply app/models/user.rb:3]: user directly involved to standard, and subject through standard

**Pull request merged**